Sources/NIOExtras/WritePCAPHandler.swift

```diff
@@ -808,6 +808,131 @@ extension NIOWritePCAPHandler {
             }
         }
     }
+
+    public final class AsynchronizedFileSink {
+        private let fileHandle: NIOFileHandle
+        private let eventLoop: EventLoop
+        private let errorHandler: (Swift.Error) -> Void
+        private var state: State = .running
+        
+        public enum FileWritingMode {
+            case appendToExistingPCAPFile
+            case createNewPCAPFile
+        }
+        
+        public struct Error: Swift.Error {
+            public var errorCode: Int
+
+            internal enum ErrorCode: Int {
+                case cannotOpenFileError = 1
+                case cannotWriteToFileError
+            }
+        }
+        
+        private enum State {
+            case running
+            case error(Swift.Error)
+        }
+        
+        /// Creates an AsynchronizedFileSink for writing to a .pcap file at the given path.
+        /// If fileWritingMode is `.createNewPCAPFile`, a file header is written.
+        public static func fileSinkWritingToFile(
+            path: String,
+            fileWritingMode: FileWritingMode = .createNewPCAPFile,
+            errorHandler: @escaping (Swift.Error) -> Void,
+            on eventLoop: EventLoop
+        ) async throws -> AsynchronizedFileSink {
+            let oflag: CInt = fileWritingMode == .createNewPCAPFile ? (O_TRUNC | O_CREAT) : O_APPEND
+            let fd: CInt = try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<CInt, Swift.Error>) in
+                path.withCString { pathPtr in
+                    let fd: Int32 = open(pathPtr, O_WRONLY | oflag, 0o600)
+                    if fd < 0 {
+                        continuation.resume(throwing: Error(errorCode: Error.ErrorCode.cannotOpenFileError.rawValue))
+                    } else {
+                        continuation.resume(returning: fd)
+                    }
+                }
+            }
+            
+            if fileWritingMode == .createNewPCAPFile {
+                _ = try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<CInt, Swift.Error>) in
```

Comment:
Same note here, this isn't actually asynchronous and doesn't need a continuation.